### PR TITLE
Add feature to edit instance group

### PR DESCRIPTION
**Pull request by nixocio**
Add feature to edit instance group.

See: https://github.com/ansible/awx/issues/7767

image

### CODE REVIEW

File: awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx [hunk 1]
```diff
@@ -11,15 +11,17 @@ import { required, minMaxValue } from '../../../util/validators';
...
     <>
       <FormField
+        name="name"
         id="instance-group-name"
         label={i18n._(t`Name`)}
-        name="name"
         type="text"
         validate={required(null, i18n)}
         isRequired
+        isDisabled={instanceGroupNameField.value === 'tower'}
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx:24]: In addition to the validation ranges you've provided, please add a min & max value for the inputs - They're supported and improve the UX when setting these values with a mouse wheel or keyboard arrows:

Screenshot from 2020-08-11 09-33-22

File: awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx:1]: @nixocio Since IGs don't have an add subscreen, this component isn't actually shared. I think we should move this file to `InstanceGroup/InstanceGroupEdit` and delete the `shared` folder (for now, that is. Container Groups may actually have a shared form but I presume you'll do that in a different PR)

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

File: awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx (hunk 1 above)
Comment (nixocio) [reply awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx:24]: Thanks for your review @jakemcdermott. Updating now.

File: awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx (hunk 2 above)
Comment (nixocio) [reply awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx:1]: `InstanceGroupForm` is shared between the `InstanceGroupAdd` and `InstanceGroupEdit` even in this case, is your suggestion still valid? I was trying to follow the same pattern that already in code basis. But it seems to be valid just in cases we have a shared `subscreen`. I will evaluate this change.

Comment (jakemcdermott) [reply awx/ui_next/src/screens/InstanceGroup/shared/InstanceGroupForm.jsx:1]: @nixocio Whoops, you're right. IGs _do_ have an add form. Never mind! It is fine as is.

**Review by jakemcdermott — APPROVED** [review]
